astropy/coordinates/representation.py

```diff
@@ -1832,15 +1832,30 @@ def _combine_operation(self, op, other, reverse=False):
     def transform(self, matrix):
         """Radial representations cannot be transformed by a Cartesian matrix.
 
+        Parameters
+        ----------
+        matrix : array-like
+            The transformation matrix in a Cartesian basis.
+            Must be a multiplication: a diagonal matrix with identical elements.
+            Must have shape (..., 3, 3), where the last 2 indices are for the
+            matrix on each other axis. Make sure that the matrix shape is
+            compatible with the shape of this representation.
+
         Raises
         ------
         NotImplementedError
+            If the matrix is not a multiplcation.
 
         """
-        raise NotImplementedError(
-            "Radial representations cannot be transformed "
-            "by a Cartesian matrix."
-        )
+        scale = matrix[..., 0, 0]
+        # check that the matrix is a scaled identity matrix on the last 2 axes.
+        if np.any(matrix != scale[..., None, None] * np.identity(3)):
+            raise NotImplementedError(
+                "Radial representations cannot be transformed"
+                " by a non-diagonal Cartesian matrix.")
```

Comment:
Update to "Radial representations can only be transformed by a scaled identity matrix``